### PR TITLE
Exporting mechs!

### PR DESCRIPTION
**Pull request by RandomMarine**
:cl: RandomMarine
add: Cargo can now export mechs!
/:cl:

Why: GOTTA GET THAT MONEY

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -56,7 +56,7 @@
 	var/datum/gas_mixture/cabin_air
 	var/obj/machinery/atmospherics/components/unary/portables_connector/connected_port = null
 
-	var/obj/item/device/radio/radio = null
+	var/obj/item/device/radio/mech/radio = null
```
Comment (Cyberboss) [inline code/game/mecha/mecha.dm]: Remove ` = null`

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -145,7 +146,7 @@
 /obj/mecha/get_cell()
 	return cell
 
-/obj/mecha/Destroy()
+/obj/mecha/deconstruct(disassembled = TRUE)
```
Comment (Cyberboss) [inline code/game/mecha/mecha.dm]: Please fully explain this change, because this makes absolutely zero sense

File: code/game/mecha/mecha_defense.dm [hunk 3]
```diff
@@ -117,7 +117,13 @@
 	if(prob(deflect_chance))
 		severity++
 		log_append_to_last("Armor saved, changing severity to [severity].")
-	. = ..()
```
Comment (Cyberboss) [inline code/game/mecha/mecha_defense.dm]: Why did you change this

**Comment by GunHog** [comment]
>Build a fully kitted out Ripley for the miners to use...
>QM steals and sells it instead

A lot of delicious salt could result from this!

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/mecha/mecha.dm (hunk 2 above)
Comment (RandomMarine) [inline code/game/mecha/mecha.dm]: deconstruct calls qdel which calls destroy
Without this change, exporting mechs makes them explode into ruined mechs once they arrive at central.

File: code/game/mecha/mecha_defense.dm (hunk 3 above)
Comment (RandomMarine) [inline code/game/mecha/mecha_defense.dm]: To preserve current behavior of severity 1 explosions destroying the mech instead of deleting it into nothingness.

**Comment by Bawhoppen** [comment]
where did gunhog's post go I was going to laugh on it

**Comment by octareenroon91** [comment]
What do you think of an export reward for ruined mechs?

File: code/game/mecha/mecha.dm (hunk 2 above)
Comment (Cyberboss) [inline code/game/mecha/mecha.dm]: That may be but Destroy does a lot of cleanup that will be needed if deconstruct isn't called. Instead just make a var that determines whether or not a mech will leave debris and set it to FALSE on export

Comment (Cyberboss) [inline code/game/mecha/mecha.dm]: In fact there already is one, just set `wreckage` to `FALSE`

File: code/game/mecha/mecha_defense.dm (hunk 3 above)
Comment (Cyberboss) [inline code/game/mecha/mecha_defense.dm]: See my above comments

File: code/modules/shuttle/supply.dm [hunk 4]
```diff
@@ -107,8 +107,11 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 	var/msg = ""
 	var/sold_atoms = ""
 
+	for(var/obj/mecha/ME in areaInstance)
+		ME.wreckage = null // So mechs don't explode when they arrive at centcom
```
Comment (RandomMarine) [inline code/modules/shuttle/supply.dm]: This feels kind of snowflaky here but I tried making the mecha export datums all within a subtype which did this on sell_object() but it never worked in testing. Or I fucked something up in my sleep deprived state.

Comment (ExcessiveUseOfCobblestone) [inline code/modules/shuttle/supply.dm]: make a subtype for the mechs and have the extra check on applies_to()

**Comment by GunHog** [comment]
> where did gunhog's post go I was going to laugh on it

It appeared to have been a double-post, so I deleted one, and it ended up removing both :x:

**Comment by Bawhoppen** [comment]
oh I thought you had just realised afterwards it was better to hide that technique so you could use it ingame without anyone knowing or blaming you

**Comment by Jordie0608** [comment]
There appears to be no checks for if a mech is occupied.

**Comment by ExcessiveUseOfCobblestone** [comment]
I think it already does a content check @Jordie0608 or else you could hide in a locker and the shuttle would take off.

**Comment by Jordie0608** [comment]
Possibly? It'd make sense.

**Comment by Tacolizard** [comment]
i mean this feature will never be used but why not

File: code/modules/shuttle/supply.dm (hunk 4 above)
Comment (RandomMarine) [inline code/modules/shuttle/supply.dm]: applies_to() isn't really doing it either

**Review by Cyberboss — APPROVED** [review]
